src/jit/codegenlegacy.cpp

```diff
@@ -18830,15 +18830,16 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                                                    (ssize_t)stubAddr);
                             // The stub will write-back to this register, so don't track it
                             regTracker.rsTrackRegTrash(compiler->virtualStubParamInfo->GetReg());
-                            getEmitter()->emitIns_R_R_I(INS_ldr, EA_PTRSIZE, REG_JUMP_THUNK_PARAM,
+                            regNumber indReg = regSet.rsGrabReg(RBM_ALLINT & ~compiler->virtualStubParamInfo->GetRegMask());
```

Comment:
The predictor change needs to be made around here https://github.com/dotnet/coreclr/blob/89d8be12677a5d395d1ec7e62874a4476ea5d4d6/src/jit/regalloc.cpp#L4876, to add an additional register requirement for this case.